src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -28,111 +61,921 @@ ActorInit En_Lift_Nuts_InitVars = {
+                    this->unk_354++;
+                }
+                break;
+        }
+    }
+    func_80AEB974(this);
+}
+
+void func_80AEAEAC(EnLiftNuts* this) {
+    this->actor.speedXZ = 2.0f;
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 1);
+    func_80AE9AC4(this, 1);
+    func_80AE9B4C(1, 1);
+    this->actionFunc = func_80AEAF14;
+}
+
+void func_80AEAF14(EnLiftNuts* this, PlayState* play) {
+    f32 dist;
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 10, 0x1000, 0x500);
+    dist = Math_Vec3f_StepTo(&this->actor.world.pos, &this->vec_1D8, this->actor.speedXZ);
+    this->actor.world.pos.y += this->actor.gravity;
+
+    if (dist == 0.0f) {
+        func_80AEAF8C(this);
+    }
+}
+
+void func_80AEAF8C(EnLiftNuts* this) {
+    this->actionFunc = func_80AEAFA0;
+}
+
+void func_80AEAFA0(EnLiftNuts* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    f32 dist;
+    f32 fv0;
+    s16 yaw;
+    s16 sp28;
+    s16 yawDiff;
+
+    sp28 = this->actor.yawTowardsPlayer - 0x8000;
+    yaw = Math_Vec3f_Yaw(&player->actor.world.pos, &this->actor.home.pos);
+    yawDiff = yaw - sp28;
+    dist = Math_Vec3f_DistXZ(&player->actor.world.pos, &this->actor.home.pos);
+
+    if (this->actor.xzDistToPlayer < dist) {
+        if (ABS_ALT(yawDiff) < 0x2000) {
+            yaw = (yawDiff > 0) ? (yaw + 0x2000) : (yaw - 0x2000);
+        }
+    }
+    if (dist < 5.0f) {
+        fv0 = 10.0f;
+    } else if (dist < 30.0f) {
+        fv0 = 40.0f;
+    } else {
+        fv0 = 80.0f;
+    }
+
+    play->actorCtx.unk268 = 1;
+    func_800B6F20(play, &play->actorCtx.unk_26C, fv0, yaw);
+    if (dist < 5.0f) {
+        func_80AEA0B4(this);
+    }
+}
+
+void func_80AEB114(EnLiftNuts* this) {
+    func_801A2BB8(NA_BGM_TIMED_MINI_GAME);
+    this->actionFunc = func_80AEB148;
+}
+
+void func_80AEB148(EnLiftNuts* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (player->stateFlags3 & PLAYER_STATE3_200) {
+        this->actor.speedXZ = 2.0f;
+        gSaveContext.eventInf[3] |= 0x10;
+        Interface_StartTimer(4, 0);
+        func_80AE9B4C(1, 2);
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_FOUND);
+        func_80AEB280(this);
+    }
+}
+
+void func_80AEB1C8(EnLiftNuts* this) {
+    this->actor.speedXZ = 2.0f;
+    gSaveContext.eventInf[3] |= 0x10;
+    Interface_StartTimer(4, 0);
+    func_80AE9B4C(1, 2);
+    this->actionFunc = func_80AEB230;
+}
+
+void func_80AEB230(EnLiftNuts* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (gSaveContext.timerCurTimes[TIMER_ID_MINIGAME_2] > 0) {
+        player->stateFlags1 &= ~PLAYER_STATE1_20;
+        func_80AEB280(this);
+    }
+}
+
+void func_80AEB280(EnLiftNuts* this) {
+    this->actionFunc = func_80AEB294;
+}
+
+void func_80AEB294(EnLiftNuts* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (((player->actor.bgCheckFlags & 1) && (player->actor.floorBgId == BG_ACTOR_MAX) &&
```

Comment:
I think this is probably BG_SCENE or whatever it's called, but there's not a lot of point changing it just here.